qiita_db/util.py

```diff
@@ -106,6 +107,8 @@ def typecast_string(string):
     The function first tries to convert to an int. If that fails, it tries to
     convert to a float. If that fails it returns the original string.
     """
+    if isinstance(string, datetime):
```

Comment:
This still makes me twitch, but the below only has a single return and a single, non-nested try/except

```python
if isinstance(string, datetime):
    item = str(string)
else:
    for fn in (int, float, str):
        try:
            item = fn(string)
        except ValueError:
            continue
        else:
            break
    return item
```